Eugene/MovieSearch/my-app/src/components/MainPage/sort-panel/SortPanel.tsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import "./SortPanel.css";
+import { parse } from "querystring";
+import { withRouter } from "react-router";
+import { RouteComponentProps } from "react-router-dom";
+
+interface ISortPanelProps extends RouteComponentProps {
+  movieCount: number;
+  handleSort(sortBy: SortBy): void;
+}
+
+interface ISortPanelState {
+  sortBy: SortBy;
+}
+
+enum SortBy {
+  Rating = "vote_average",
+  ReleaseDate = "release_date",
+}
+
+class SortPanel extends React.Component<ISortPanelProps, ISortPanelState> {
+  state: ISortPanelState = {
+    sortBy: SortBy.Rating,
+  };
+
+  componentDidMount() {
+    const searchParams = this.props.location.search.slice(1);
+    const parsed = parse(searchParams) as {
+      sortBy: string;
+    };
+
+    if (parsed.sortBy) {
+      if (parsed.sortBy === SortBy.Rating) {
+        this.setState({ sortBy: SortBy.Rating });
+      } else {
+        this.setState({ sortBy: SortBy.ReleaseDate });
+      }
+    }
+  }
+
+  componentDidUpdate(prevprops: RouteComponentProps) {
+    if (this.props.location.search !== prevprops.location.search) {
+      const searchParams = this.props.location.search.slice(1);
+      const parsed = parse(searchParams) as {
+        sortBy: string;
+      };
+
+      if (parsed.sortBy) {
+        if (parsed.sortBy === SortBy.Rating) {
+          this.setState({ sortBy: SortBy.Rating });
+        } else {
+          this.setState({ sortBy: SortBy.ReleaseDate });
+        }
+      }
+    }
+  }
```

Comment:
It is seems like you have to update your check a bit. You shouldn't update state when you pushed a new value in the URL. Because in this case you gonna setState twice. First, when user clicked on SortBy button and then when ** handleSort.** fired and url changed.